src/Dapr.Workflow/WorkflowMetadata.cs

```diff
@@ -0,0 +1,44 @@
+﻿// ------------------------------------------------------------------------
+// Copyright 2022 The Dapr Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using Microsoft.DurableTask.Client;
+
+namespace Dapr.Workflow
+{
+    /// <summary>
+    /// Represents a snapshot of a workflow instance's current state, including metadata.
+    /// </summary>
+    public class WorkflowMetadata
+    {
+        internal WorkflowMetadata(OrchestrationMetadata? metadata)
```

Comment:
Nit: `details` instead of `metadata`?

Reply:
Metadata is a common idiom on Dapr, even on the .Net SDK. While I don't have strong opinions about this, I just want to make sure this is to align with .Net SDKs expectations or more a matter of preference.

Reply:
I'm fine either way on this too. If "metadata" is too overloaded, then I think "details" is fine.

Reply:
Metadata and Options are both used in the SDK. I'm fine either way, but would at least prefer the member to match as it feels weird to me to have something like `workflowMetadata.Details` and would prefer to either have `workflowDetails` or `workflowMetadata.OrchestrationMetadata`.

Reply:
Marking this as resolved